proto/scheduler.proto

```diff
@@ -101,12 +101,38 @@ message EnqueueTaskReservationRequest {
 }
 
 message EnqueueTaskReservationResponse {
-  // Intentionally left blank.
+  string task_id = 1;
+}
+
+message RegisterExecutorRequest {
+  ExecutionNode node = 1;
 }
 
-service Scheduler {
+message ExecutorToSchedulerMessage {
```

Comment:
Agree that I like having the rpc match the message. 

Either naming this outer proto RegisterExecutorRequest (since it’s the RegisterExecutor rpc) and then you renaming the inner protos to something like RegistrationInfo and TaskReservation + TaskReservationAck

Or if this stream really encompasses all communication between executor and scheduler, you could call the rpc something like ExecutorStream/ExecutionStream/ReservationStream/RegistrationStream and then have ExecutorStreamRequest and ExecutorStreamResponse. 

Idk, I don't feel super strongly about this stuff though, just feels a little confusing.

Reply:
Renamed RPC to `RegisterAndStreamWork` and messages to `RegisterAndStreamWorkRequest` and `RegisterAndStreamWorkResponse`.